src/main/java/org/spongepowered/api/event/resource/PackEvent.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.event.resource;
+
+import org.spongepowered.api.event.Event;
+import org.spongepowered.api.resource.pack.MutablePackList;
+import org.spongepowered.api.resource.pack.Pack;
+import org.spongepowered.api.resource.pack.PackDiscoverer;
+import org.spongepowered.api.resource.pack.PackList;
+
+/**
+ * Base {@link Pack} event.
+ */
+public interface PackEvent extends Event {
+
+    /**
+     * Gets the relevant {@link PackList} for this event.
+     *
+     * @return The pack list
+     */
+    PackList getPackList();
+
+    /**
+     * Event for registering {@link PackDiscoverer}s. It is fired sometime during init.
+     */
+    interface RegisterPackDiscoverer extends PackEvent {
+
+        /**
+         * {@inheritDoc}
+         * <p>Returns a {@link MutablePackList} to expose methods for adding listeners.</p>
+         * @return The pack list
+         */
+        @Override
+        MutablePackList getPackList();
```

Comment:
Do not expose the pack list, instead add a register method like the registry events.